### ISSUE TITLE
Terminators should print out what they're doing

### ISSUE THREAD

**Issue by bwspenc** [idaholab/moose#20565]
## Reason
Currently, if a Terminator object does a hard or soft termination, nothing is printed out to the console to alert the user to what's going on. This can result in users that are very puzzled about why their job is terminating if working with a model they're not familiar with.

## Design
A simple message that says something like: "Hard|Soft termination of job due to terminator <terminator_name>" would be very helpful. It looks like there's actually an option in the Terminator class for the user to supply a message, but I'm not sure why anyone would think to manually add that. Maybe the intent is for derived classes to override that? Regardless of the mechanism, we should be printing something out by default.

I would propose changing this so that if a user provided the `message` parameter, we would use that, but if not, we would generate a simple default message.

## Impact
This should have minimal impact. We would start seeing this message in output files, and would honor the `message` parameter if supplied.

**Comment by dschwen** [comment]
That functionality already exists:

https://github.com/idaholab/moose/blob/next/framework/src/userobject/Terminator.C#L37-L45

**Comment by dschwen** [comment]
Ah, ok, default message is fine at INFO level

**Comment by bwspenc** [comment]
No, my point is that is shouldn't be through an optionally supplied message. Nobody is going to think to specify that.

**Comment by bwspenc** [comment]
I'm really curious what the use case for a user-supplied message is. If anyone knows why it was designed this way, I'd be interested to know. I just spent a while helping someone figure out why their model was silently ending in the middle of the analysis. We were at a total loss, until by chance I remembered that the Terminator acts this way.

**Comment by bwspenc** [comment]
Makes sense. However, it seems like a generic message like "Terminator 'precipitation_count_limit' caused a hard failure" would probably meet that need most of the time. We can keep the option to override that message with a custom one, but I can't imagine that it would get used that much if we print a short message that conveys the key information by default.
